clang/lib/CConv/CheckedRegions.cpp

```diff
@@ -187,6 +174,90 @@ bool CheckedRegionFinder::VisitCStyleCastExpr(CStyleCastExpr *E) {
   return true;
 }
 
+bool CheckedRegionFinder::VisitCallExpr(CallExpr *C) {
+  // TODO this should really be moved to the other visitor
+  // both for clarity and because it will allow it take advantage of
+  // redundancy checks
+  auto FD = C->getDirectCallee();
+  FoldingSetNodeID ID;
+  C->Profile(ID, *Context, true);
+  if (FD && FD->isVariadic()) {
+    Wild = true;
+  } else {
+    if (FD) {
+      auto type = FD->getReturnType();
+      Wild |= (!(FD->hasPrototype() || FD->doesThisDeclarationHaveABody()))
+        || containsUncheckedPtr(type)
+        || (any_of(FD->param_begin(), FD->param_end(), [this] (Decl *param) {
+              auto CVSet = Info.getVariable(param, Context);
+              return isWild(CVSet );
+            }));
+    }
+    handleChildren(C->children());
+    Map[ID] = Wild ? IS_UNCHECKED : IS_CHECKED;
```

Comment:
Should this assignment to `Map[ID]` be outside the conditional? What if the true branch is taken? You are setting `Wild` to true on line 185 but then you are not updating `Map`.

Reply:
Yup, this is a mistake.